From aef862b0b2076720b6c651997516a0698d066e67 Mon Sep 17 00:00:00 2001
From: Hugues Kamba-Mpiana <hugues.kambampiana@arm.com>
Date: Wed, 3 Jan 2024 14:16:58 +0000
Subject: [PATCH] Add non-constant appFirmwareVersion extern

The const qualifier version of appFirmwareVersion is not compatible
with freertos_ota_pal_psa definition

Signed-off-by: Hugues Kamba-Mpiana <hugues.kambampiana@arm.com>
---
 source/include/ota_appversion32.h | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/source/include/ota_appversion32.h b/source/include/ota_appversion32.h
index 61ee83b..30b6f32 100644
--- a/source/include/ota_appversion32.h
+++ b/source/include/ota_appversion32.h
@@ -75,7 +75,11 @@ typedef struct
     } u; /*!< @brief Version based on configuration in big endian or little endian. */
 } AppVersion32_t;
 
-extern const AppVersion32_t appFirmwareVersion; /*!< @brief Making the version number available globally through external linkage. */
+#if ( defined( OTA_USE_NONCONST_APPVERSION ) && ( OTA_USE_NONCONST_APPVERSION == 1 ) )
+    extern AppVersion32_t appFirmwareVersion;       /*!< @brief Making the version number available globally through external linkage, without const qualifier. */
+#else
+    extern const AppVersion32_t appFirmwareVersion; /*!< @brief Making the version number available globally through external linkage, with const qualifier. */
+#endif
 
 /* *INDENT-OFF* */
 #ifdef __cplusplus
-- 
2.34.1

